src/internal/connector/onedrive/collection.go

```diff
@@ -173,12 +173,69 @@ func NewCollection(
 		return nil, clues.Wrap(err, "getting previous location").With("prev_path", prevPath.String())
 	}
 
+	c := newColl(
+		itemClient,
+		folderPath,
+		prevPath,
+		driveID,
+		service,
+		statusUpdater,
+		source,
+		ctrlOpts,
+		colScope,
+		doNotMergeItems)
+
+	c.locPath = locPath
+	c.prevLocPath = prevLocPath
+
+	return c, nil
+}
+
+// NewCollectionPrefix creates a Collection prefix, without parsing a location path.
+// Should not be used unless the paths are explicitly known to be prefixes.
+func NewCollectionPrefix(
```

Comment:
since we switched to using the graph package's prefix collection I don't think this function should be required anymore?